apps/src/code-studio/components/progress/StageLockDialog.jsx

```diff
@@ -0,0 +1,284 @@
+    // (i.e. so that checkbox is updated) before we call setState and rerender
+    setTimeout(() => {
+      this.setState({
+        lockStatus: this.state.lockStatus.map((item, index) => {
+          if (index !== modifiedIndex) {
+            return item;
+          }
+          return Object.assign({}, item, {
+            lockStatus: value
+          });
+        })
+      });
+    }, 0);
+  },
+
+  handleSave() {
+    this.props.saveDialog(this.state.lockStatus);
+  },
+
+  render() {
+    return (
+      <BaseDialog
+        isOpen={this.props.isOpen}
+        handleClose={this.props.handleClose}
+      >
+        <div style={styles.main}>
+          <div style={styles.title}>{commonMsg.assessmentSteps()}</div>
+          <table>
+            <tbody>
+              <tr>
+                <td>{commonMsg.allowEditingInstructions()}</td>
+                <td>
+                  <button
+                    style={progressStyles.orangeButton}
+                    onClick={this.allowEditing}
+                  >
+                    {commonMsg.allowEditing()}
+                  </button>
+                </td>
+              </tr>
+              <tr>
+                <td>{commonMsg.lockStageInstructions()}</td>
+                <td>
+                  <button
+                    style={progressStyles.orangeButton}
+                    onClick={this.lockStage}
+                  >
+                    {commonMsg.lockStage()}
+                  </button>
+                </td>
+              </tr>
+              <tr>
+                <td>{commonMsg.showAnswersInstructions()}</td>
+                <td>
+                  <button
+                    style={progressStyles.orangeButton}
+                    onClick={this.showAnswers}
+                  >
+                    {commonMsg.showAnswers()}
+                  </button>
+                </td>
+              </tr>
+              <tr>
+                <td>{commonMsg.relockStageInstructions()}</td>
+                <td>
+                  <button
+                    style={progressStyles.orangeButton}
+                    onClick={this.lockStage}
+                  >
+                    {commonMsg.relockStage()}
+                  </button>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+          <div style={styles.descriptionText}>{commonMsg.autolock()}</div>
+          <div style={styles.title}>{commonMsg.studentControl()}</div>
+          <div style={styles.descriptionText}>
+            {commonMsg.studentLockStateInstructions()}
+          </div>
+          <table style={styles.studentTable}>
+            <tbody>
+              <tr>
+                <th style={styles.headerRow}>{commonMsg.student()}</th>
+                <th style={styles.headerRow}>{commonMsg.locked()}</th>
+                <th style={styles.headerRow}>{commonMsg.editable()}</th>
+                <th style={styles.headerRow}>{commonMsg.answersVisible()}</th>
+              </tr>
```

Comment:
Why is your header row inside `tbody`?  More generally, have you seen [`Reactabular`](http://reactabular.js.org/)?